Tests/MongoSwiftTests/ClientSessionTests.swift

```diff
@@ -293,4 +308,207 @@ final class ClientSessionTests: MongoSwiftTestCase {
+                    return
+                }
+                expect((event.command["readConcern"] as? Document)?["afterClusterTime"]).to(beNil())
+                seenCommand = true
+            }
+            defer { center.removeObserver(observer) }
+            _ = try collection.aggregate([["$match": ["x": 1] as Document]], session: session).next()
+            expect(seenCommand).to(beTrue())
+        }
+
+        // spec test 9
+        try client.withSession(options: ClientSessionOptions(causalConsistency: true)) { session in
+            let collection1 = db.collection(self.getCollectionName(),
+                                            options: CollectionOptions(readConcern: ReadConcern(.snapshot)))
+            _ = try collection1.find(session: session).next()
+            let opTime = session.operationTime
+
+            var seenCommand = false
+            let observer = center.addObserver(forName: .commandStarted, object: nil, queue: nil) { notif in
+                guard let event = notif.userInfo?["event"] as? CommandStartedEvent else {
+                    return
+                }
+                let readConcern = event.command["readConcern"] as? Document
+                expect(readConcern).toNot(beNil())
+                expect(readConcern!["afterClusterTime"]).to(bsonEqual(opTime))
+                expect(readConcern!["level"]).to(bsonEqual("snapshot"))
+                seenCommand = true
+            }
+            defer { center.removeObserver(observer) }
+            _ = try collection1.find(session: session).next()
+            expect(seenCommand).to(beTrue())
+        }
+
+        // spec test 10
+        try client.withSession(options: ClientSessionOptions(causalConsistency: true)) { session in
+            let collection1 = db.collection(self.getCollectionName(),
+                                            options: CollectionOptions(writeConcern: try WriteConcern(w: .number(0))))
+            try collection1.insertOne(["x": 3])
+            expect(session.operationTime).to(beNil())
+        }
+
+        // spec test 12
+        try client.withSession(options: ClientSessionOptions(causalConsistency: true)) { session in
+            var seenCommand = false
+            let observer = center.addObserver(forName: .commandStarted, object: nil, queue: nil) { notif in
+                guard let event = notif.userInfo?["event"] as? CommandStartedEvent else {
+                    return
+                }
+                expect(event.command["$clusterTime"]).toNot(beNil())
+                seenCommand = true
+            }
+            defer { center.removeObserver(observer) }
+            _ = try collection.find(session: session).next()
+            expect(seenCommand).to(beTrue())
+        }
+    }
+
+    /// Test causal consistent behavior on a topology that doesn't support cluster time.
+    func testCausalConsistencyStandalone() throws {
```

Comment:
it seems like tests 1, 2, and 10 are also ok to run without cluster time, is it worth doing that?

Reply:
Its only a little copypaste, so sure. done